_posts/en/2024-10-09-wallet-integration-guide.md

```diff
@@ -0,0 +1,590 @@
+for valueless anchors.
+
+Instead, let us try to fit the required transaction structure into the 1P1C
+paradigm as best we can, at the cost of some additional fees. All Ark tree
+transactions, starting at the root, are a TRUC transaction and add to it a
+minimal satoshi value P2A output.
+
+When a transactor chooses to unilaterally exit from an Ark, the user publishes
+the root transaction plus spend of P2A for fees, then waits for confirmation.
+Once confirmed, the user submits the next transaction in their merkle branch,
+along with its own P2A being spend for CPFP. This continues until the whole
+merkle branch is published and the funds are safely extracted from the Ark tree.
+Other users of the same Ark may maliciously or accidentally submit the same
+inner node transaction with too low a feerate, but sibling eviction would ensure
+that at every step honest child transactions below 1kvB could RBF competing
+children without requiring all other outputs be locked, or multiple anchors.
+
+Assuming binary trees, this comes at the asymptotic cost of nearly 100% vbyte
+overhead versus the idealized Ark for the first user, and roughly 50% over the
+whole tree if it completely unrolls. For 4-nary, this diminishes to ~25% over
+the whole tree.
+
+### LN Splices
+
+Other topologies also come up in more advanced Lightning Network constructs that
+may need some work to match with 1P1C relay.
+
+Lightning Network [splices][topic splicing] are an emerging standard and in general use already.
+Each splice is a spending of the original funding output, re-depositing contract
+funds into a new funding output, with the same presigned commitment transaction
+chain as before. While unconfirmed, the original channel state and new channel
+state(s) are simultaneously signed and tracked.
+
+One example that could exceed the 1P1C paradigm is the case where:
+
+1. Alice and Bob fund a channel.
+2. Alice splices *out* some funds to an on-chain address controlled by Carol,
+who is using a cold set of keys, so she cannot CPFP. This splice-out is shooting
+for confirmation within hours.
+3. Bob’s machine shuts down, or node force closes for some reason as too-low
+feerate.
+4. A new BRC-20 issuance occurs and feerates skyrocket, delaying confirmation of
+the splice-out transaction unduly.
+
+Alice wants the on-chain payment to Carol happen, so she does not go to chain
+with a commitment transaction without the splice. This means
+splice_tx->commitment_tx->anchor_spend becomes the required package to make this
+propagate.
+
+Instead, let us consider how to fit it within the 1P1C paradigm, without unduly
+wasting vbytes when unnecessary. A LN wallet could instead of running one
+splice-out per on-chain payment, could do 2 splice-outs, since they’re
+conflicting. One version uses the relatively conservative feerate chosen by the
+fee estimator. The other version could include a P2A output at 240 satoshis, or
+0 satoshis in the future with Ephemeral Dust.
```

Comment:
hmmm might be less confusing to leave this out for now... and we can do a 29.0 wallet integration guide? With Ephemeral Dust, LN symmetry, blackjack, etc?

Reply:
A mention somewhere might be helpful for people who are confused how Ephemeral Anchors disappeared as a single idea.